include/vcpkg/vcpkgcmdarguments.h

```diff
@@ -71,13 +73,69 @@ namespace vcpkg
         MetadataMessageKind kind;
         union
         {
-            char unused;
             const msg::MessageT<>* message;
-            StringLiteral literal;
+            const char* literal; // not StringLiteral so this union is sizeof(void*)
             LocalizedString (*callback)();
         };
     };
 
+    inline constexpr bool constexpr_contains(const char* haystack, const char* needle) noexcept
+    {
+        for (;; ++haystack)
+        {
+            for (std::size_t offset = 0;; ++offset)
+            {
+                if (!needle[offset])
+                {
+                    return true;
+                }
+
+                if (!haystack[offset])
+                {
+                    return false;
+                }
+
+                if (needle[offset] != haystack[offset])
+                {
+                    break;
+                }
+            }
+        }
+    }
+
+    static_assert(constexpr_contains("", ""), "boom");
+    static_assert(constexpr_contains("hay", ""), "boom");
+    static_assert(!constexpr_contains("", "needle"), "boom");
+    static_assert(constexpr_contains("needle", "nee"), "boom");
+
+    struct LearnWebsiteLinkLiteralUndocumentedCookie
+    {
+    };
+
+    constexpr LearnWebsiteLinkLiteralUndocumentedCookie Undocumented;
+
+    struct LearnWebsiteLinkLiteral
+    {
+        /*implicit*/ constexpr LearnWebsiteLinkLiteral(LearnWebsiteLinkLiteralUndocumentedCookie) noexcept : literal{}
+        {
+        }
+
+        template<int N>
+        /*implicit*/ constexpr LearnWebsiteLinkLiteral(const char (&literal)[N]) noexcept : literal(literal)
+        {
+            assert(!constexpr_contains(literal, "en-us") &&
```

Comment:
Is there some sort of divide-by-zero or array-of-zero thing that can be done here to make this assert compile time?

Reply:
It is compile time; `assert` is constexpr when the assertion passes and not constexpr when the assertion fails.